lib/module.js

```diff
@@ -53,8 +53,10 @@ export default function SentryModule (moduleOptions) {
     }
   }
 
+  const privateRuntimeConfig = this.options.privateRuntimeConfig || {}
+  const runtimeSentryConfig = (privateRuntimeConfig && privateRuntimeConfig.sentry) || {}
```

Comment:
This is not going to work. `privateRuntimeConfig` is what is going to be available in generated plugin (in **run**-time), not in module (in **build**-time). You were probably misled by the fact that on developer machine, the environment is the same/similar during build and run. This is totally different in production.

You need to some like this in module:

```ts
  const defaults = {
    useRuntimeConfig: this.options.publicRuntimeConfig ? 'sentry' : undefined,
    ...
```

and this in plugin:

```ts
export default ({ app: { router }, $config }) => {

  const options = <%= JSON.stringify(options) %>
  <% if (options.useRuntimeConfig) { %>
    Object.assign(options, $config[options.useRuntimeConfig])
  <% } %>
  // You may want to remove `useRuntimeConfig` from the options object
  // if you're passing it directly to Sentry.
```

This will be compatible with Nuxt <2.13, and it will allow to override runtime config option key or disable it completely.